mt-kahypar/partition/refinement/label_propagation/label_propagation_refiner.cpp

```diff
@@ -278,7 +275,13 @@ namespace mt_kahypar {
     timer.stop_timer("rebalance_lp");
     DBG << "[LP] Imbalance after rebalancing: " << current_metrics.imbalance << ", quality: " << current_metrics.quality;
 
-    if (current_metrics.quality > best_metrics.quality) { // rollback and stop LP
+    bool was_imbalanced_and_improved_balance = !_old_partition_is_balanced
+                                               && current_metrics.imbalance < best_metrics.imbalance;
+    bool is_imbalanced_and_worsened_balance = (!metrics::isBalanced(hypergraph, _context)
+                                              && current_metrics.imbalance > best_metrics.imbalance);
+    if (!was_imbalanced_and_improved_balance
```

Comment:
I had a hard time parsing this (double negation) but it looks correct to me. A comment would be nice here.

Reply:
I think I found a simpler way to express it (see new commit)